### PR TITLE
=htc #20024 introduce more specific exception type for connection close

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/akka/akka/issues/20024

### CODE REVIEW

**Comment by johanandren** [comment]
LGTM

File: akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala [hunk 1]
```diff
@@ -237,4 +237,10 @@ private object PoolSlot {
         context.stop(self)
     }
   }
+
+  final class UnexpectedDisconnectException(msg: String, cause: Throwable) extends RuntimeException(msg, cause) {
+    def this(msg: String) {
```
Comment (patriknw) [inline akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala]: I think that we more often use this syntax:
```
def this(msg: String) = this(msg, null)
```

Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala]: yep, will use that. I so rarely write secondary constructors that I'm always confused about syntax of them ;)

**Comment by patriknw** [comment]
LGTM, aside from the nitpick

**Pull request merged**